File: shield/shieldtest.py
import unittest

from shield import Shield

class ShieldTest(unittest.TestCase):
    def testDefaultShield(self):
        style : str = "round"
        material : str = "steel"
        weight : int = 10
        color : str = "silver"
        thrown : bool = False

        shield : Shield = Shield(style, material, weight, color, thrown)
        self.assertEqual(shield.style,style)
        self.assertEqual(shield.material,material)
        self.assertEqual(shield.weight,weight)
        self.assertEqual(shield.color,color)
        self.assertEqual(shield.thrown,thrown)

    def testThrow(self):
        style : str = "round"
        material : str = "vibranium"
        weight : int = 12
        color : str = "red, white, and blue"
        thrown : bool = False

        shield : Shield = Shield(style, material, weight, color, thrown)
        self.assertEqual(shield.style,style)
        self.assertEqual(shield.material,material)
        self.assertEqual(shield.weight,weight)
        self.assertEqual(shield.color,color)
        self.assertEqual(shield.thrown,thrown)

        if shield.thrown == False:
            print("Captain America throws his mighty shield!")
        shield.throw()
        self.assertEqual(shield.thrown, True)
        
if __name__ == '__main__':
    unittest.main()

File: hw3/test/shieldtest.js
const assert = require('assert');
const chai = require('chai');
const shield = require('../hello');

const expect = chai.expect;
const assertEq = assert.deepStrictEqual

describe('Shield', function() {
  describe('#construct()', function() {
    it('shields can be made', function() {
	    const style = "round"
      const material = "steel"
      const color = "silver"
      const weight = 10
      const thrown = false

      const shield = Shield = new shield.Shield(style, material, color, weight, thrown)
      assertEq(shield.style,style)
      assertEq(shield.material,material)
      assertEq(shield.color,color)
      assertEq(shield.weight,weight)
      assertEq(shield.thrown,thrown)
    });
  });
});

describe('Shield', function() {
  describe('#construct()', function() {
    it('shields can be made', function() {
	    const style = "round"
      const material = "vibranium"
      const color = "red, white, and blue"
      const weight = 12
      const thrown = false

      const shield = Shield = new shield.Shield(style, material, color, weight, thrown)
      assertEq(shield.style,style)
      assertEq(shield.material,material)
      assertEq(shield.color,color)
      assertEq(shield.weight,weight)
      assertEq(shield.thrown,thrown)

      if (shield.thrown == false){
        console.log("Captain America throws his mighty shield!")
        shield.throwShield()
        self.assertEqual(shield.thrown, true)
      }
    });
  });
});